crates/uv/src/commands/project/sync.rs

```diff
@@ -1294,6 +1306,77 @@ impl SyncReport {
     }
 }
 
+/// A summary of a single package change performed during sync.
+#[derive(Serialize, Debug, Clone)]
+struct PackageChangeReport {
+    /// The normalized package name.
+    name: PackageName,
+    /// The resolved version of the package.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    version: Option<uv_pep440::Version>,
+    /// The source for URL-based requirements.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    source: Option<PackageChangeSourceReport>,
+    /// The action that was taken for the package.
+    action: PackageChangeAction,
+}
+
+impl PackageChangeReport {
+    fn from_changelog(changelog: &Changelog) -> Vec<Self> {
+        let mut changes: Vec<_> = changelog
+            .uninstalled
+            .iter()
+            .map(|dist| Self::from_dist(dist, PackageChangeAction::Removed))
+            .chain(
+                changelog
+                    .installed
+                    .iter()
+                    .map(|dist| Self::from_dist(dist, PackageChangeAction::Added)),
+            )
+            .chain(
+                changelog
+                    .reinstalled
+                    .iter()
+                    .map(|dist| Self::from_dist(dist, PackageChangeAction::Reinstalled)),
+            )
+            .collect();
+
+        changes.sort_by(|a, b| {
+            a.name
+                .cmp(&b.name)
+                .then_with(|| a.action.cmp(&b.action))
+                .then_with(|| a.version.cmp(&b.version))
+        });
+        changes
+    }
+
+    fn from_dist(dist: &ChangedDist, action: PackageChangeAction) -> Self {
+        Self {
+            name: dist.name().clone(),
+            version: dist.version().cloned(),
+            source: dist
+                .url()
+                .map(|url| PackageChangeSourceReport { url: url.clone() }),
+            action,
+        }
+    }
+}
+
+/// The action taken on an individual package during sync.
+#[derive(Serialize, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[serde(rename_all = "snake_case")]
+enum PackageChangeAction {
+    Removed,
+    Added,
```

Comment:
I think I'd expect "Uninstalled" and "Installed" here if we're using "Reinstalled"?